DummyFile.java

```diff
@@ -1,4 +1,40 @@
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.AmazonKinesisClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+
+import example.foo.FooClient;
+
 public class DummyFile {
-    jkgfldksghkjfdhgkfdgkfdhsjnvjkfdnvkjfdnvkjf
-        gfdgdfjkgkdj
+    private AmazonS3 s3Client = AmazonS3ClientBuilder.standard().build();
+    private AmazonKinesisClient kinesisClient = AmazonKinesisClientBuilder.defaultClient();
+    private FooClient fooClient;
+
+    // Should flag and suggest v2
+    public void s3ListObjects() {
+        s3Client.listObjects(new ListObjectsRequest());
+    }
+
+    // Should not flag
+    public void s3ListObjectsV2() {
+        s3Client.listObjectsV2(new ListObjectsV2Request());
+    }
+
+    // Should flag and suggest Transfer Manager
+    public void s3MultiPartUpload() {
+        s3Client.initiateMultipartUpload(new InitiateMultipartUploadRequest());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code manually manages uploads or downloads. The SDK can automatically handle these operations. Consider using the SDK to automatically handle this instead. SDK class: TransferManager. See [Using TransferManager for Amazon S3 Operations](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-s3-transfermanager.html).